tests/fixtures/db/__init__.py

```diff
@@ -1,21 +1,26 @@
 import logging
+import os
 import warnings
+from typing import Generator
 
 import pytest
-from _pytest.monkeypatch import MonkeyPatch  # noqa: WPS436
 from sqlalchemy import MetaData, create_engine
 from sqlalchemy.exc import SAWarning
 from sqlalchemy.orm import close_all_sessions, sessionmaker
 
 
 @pytest.fixture(scope="session", autouse=True)
-def engine_uri_env(engine_uri):
-    monkeypatch = MonkeyPatch()
-    monkeypatch.setenv("MELTANO_DATABASE_URI", engine_uri)
-    try:
+def engine_uri_env(engine_uri: str) -> None:
+    """Use the correct meltano database URI for these tests."""
+    os.environ["MELTANO_DATABASE_URI"] = engine_uri
+
+
+@pytest.fixture
+def un_engine_uri(monkeypatch) -> Generator:
```

Comment:
Yeah I understand the thinking my thought was that I'd need to explain why the side affect was happening inside of the test with a comment. This way the fixture is self describing. Both work, I'll go with your way as you're probably right that other tests probably won't need to do this

Reply:
We have to set the `MELTANO_DATABASE_URI`  before the `project` fixture is called because in `/home/visch/git/meltano/src/meltano/core/db.py(26)` (below) 
```
> /home/visch/git/meltano/src/meltano/core/db.py(26)project_engine()
-> if existing_engine:
(Pdb) print (existing_engine)
(Engine(sqlite:////tmp/pytest-of-visch/pytest-14/fixture_db_sqlite_engine_uri0/pytest_meltano.db), sessionmaker(class_='Session', bind=Engine(sqlite:////tmp/pytest-of-visch/pytest-14/fixture_db_sqlite_engine_uri0/pytest_meltano.db), autoflush=True, autocommit=False, expire_on_commit=True))
(Pdb) print(os.getenv("MELTANO_DATABASE_URI"))
None
```

Caused by `_engines` here
src/meltano/core/db.py
```
>>  6 _engines = {}
    5
    4
>>  3 def project_engine(project, default=False) -> tuple(Engine, sessionmaker):
    2     """Creates and register a SQLAlchemy engine for a Meltano project instance."""
    1
  24      breakpoint()
    1     existing_engine = _engines.get(project)
 ```


Doing this via a fixture makes sense because of that. I tried to make this work without `un_engine_uri` and ended up back to this :O. We could add an issue for this, I think _engines sticks around due to the `project* `fixture

More rambling and details here https://github.com/meltano/meltano/pull/6407#issuecomment-1200527808

wdyt @WillDaSilva

Reply:
Reloading a module is definitely worse - keeping `un_engine_uri` as a fixture makes sense given this context. A comment explaining that it must run before the project is created would be appreciated.

Reply:
Done!